### PR TITLE
Issue #736: Fixes unsafe cast in CreateStagingTableAnalyzer

### LINKED ISSUE #736: Unsafe cast in CreateStagingTableAnalyzer
line 107: 
```
final DatastoreCreationDelegate creationDelegate = new DatastoreCreationDelegateImpl(
                (MutableDatastoreCatalog) datastoreCatalog);
```

This is highly unsafe since datastore catalog might be e.g. DatastoreCatalogImpl.

Instead let the DatastoreCreationDelegate class do a check and adapt it's behaviour accordingly.

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes #736 - unsafe cast in CreateStagingTableAnalyzer

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/output/datastore/DatastoreCreationDelegateImpl.java [hunk 1]
```diff
@@ -29,23 +30,27 @@
...
+    public void createDatastore(Datastore datastore) {
+        final String name = datastore.getName();
+        synchronized (_datastoreCatalog) {
+            if (_datastoreCatalog.containsDatastore(name)) {
+                logger.warn("Datastore '{}' already exists. No new datastore will be created!", name);
+            } else {
+                if (_datastoreCatalog instanceof MutableDatastoreCatalog) {
+                    ((MutableDatastoreCatalog) _datastoreCatalog).addDatastore(datastore);
+                } else {
+                    logger.error("Tried to create datastore '{}', but the datastore catalog is not mutable", name);
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/output/datastore/DatastoreCreationDelegateImpl.java]: To the user, nothing really happens, as far as I can see. Shouldn't this throw an exception so the user can get a warning?

**Comment by ClaudiaPHI** [comment]
LGTM!

File: desktop/ui/src/main/java/org/datacleaner/output/datastore/DatastoreCreationDelegateImpl.java (hunk 1 above)
Comment (kaspersorensen) [inline desktop/ui/src/main/java/org/datacleaner/output/datastore/DatastoreCreationDelegateImpl.java]: Yes ... It should never happen actually. So better throw exception.

**Comment by kaspersorensen** [comment]
Let's merge then when Travis is done with it.

**Comment by LosD** [comment]
It is now. Merging!

**Pull request merged**